### PR TITLE
get/api/topics completed

### PR DESCRIPTION
**Pull request by laurie60**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const seed = require("../db/seeds/seed");
+const {
+  articleData,
+  commentData,
+  topicData,
+  userData,
+} = require("../db/data/test-data");
```
Comment (P-Copley) [inline __tests__/app.test.js:7]: You're destructuring these only to put them back into an object below. Will be shorter to not bother:

```js
const testData = require('../db/data/test-data')
```

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+const db = require("../db/connection");
+const request = require("supertest");
+const app = require("../db/app");
+const sorted = require("jest-sorted");
+
+beforeEach(() => seed({ articleData, commentData, topicData, userData }));
+// afterAll(() => db.end);
```
Comment (P-Copley) [inline __tests__/app.test.js:14]: You should keep the afterAll in but the `db.end()` should be invoked 👍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+// describe.only("Name of the group", () => {
+//   test("should ", () => {
+//     return db.query("SELECT * FROM topics").then((topics) => {
+//       console.log(topics.rows);
+//     });
+//   });
+// });
```
Comment (P-Copley) [inline __tests__/app.test.js:22]: These can be deleted 👍

File: __tests__/app.test.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  describe("get/api/topics", () => {
+    test("200: responds with array of objects, each of which have properties slug & description ", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+          body.forEach((topic) => {
+            expect(topic).toHaveProperty("slug");
+            expect(topic).toHaveProperty("description");
+          });
```
Comment (P-Copley) [inline __tests__/app.test.js:34]: As you're doing a forEach here this test will pass if the endpoint responds with an empty array. Consider adding a length check in to make sure you're not getting false positives

File: __tests__/app.test.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        });
+    });
+    test("404: responds with a message when non existant endpoint requested ", () => {
+      return request(app)
+        .get("/ipa")
```
Comment (P-Copley) [inline __tests__/app.test.js:39]:
```js
        .get("/ipa")
```

I approve of this test 😁

File: db/controller.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+const { fetchTopics } = require("./models");
+
+const { req, res } = require("./app");
+
+exports.getTopics = (req, res) => {
+  console.log(req.query, "<<<<<<<<<<<<<<<<query");
+
+  fetchTopics().then((responses) => {
+    //console.log(treasures.length);
+    res.status(200).send(responses);
```
Comment (P-Copley) [inline db/controller.js]:
```suggestion
    res.status(200).send({ topics: responses });
```

The tickets specify the data being sent back on a key of the resources name. In this case `topics` please

**Comment by P-Copley** [comment]
You've got a few comments left lying around that need tidying up but once they're done and you've had a look at the above feel free to merge 😁

**Review by laurie60 — COMMENTED** [review]
Thanks Paul!

**Pull request merged**